src/pudl/transform/vcerare.py

```diff
@@ -278,17 +331,64 @@ def _spot_fix_great_lakes_columns(df: pd.DataFrame) -> pd.DataFrame:
     )
 
 
+def _standardize_census_names(combined_df: pd.DataFrame, census_pep_data: pd.DataFrame):
+    """Make sure that the county names correspond to the latest census vintage."""
+    # Get unique list of county and place name combinations
+    county_fips = combined_df[["place_name", "county_id_fips"]].drop_duplicates()
+    county_fips = county_fips.dropna(
+        subset=["county_id_fips"]
+    )  # Drop non-FIPS place names (e.g., lakes)
+    census_fips = census_pep_data[["county_id_fips", "area_name"]].drop_duplicates()
+    census_fips = census_fips.loc[
+        census_fips.area_name.str.contains(
+            r"County$|Parish$", na=False
+        )  # Drop city and township census data
+    ]
+    # Create a map of place names from VCE to Census data
+    names_df = county_fips.merge(
+        census_fips, on="county_id_fips", how="left", validate="1:1"
+    )
+    names_df["area_name"] = names_df["area_name"].str.lower()
+    names_df["area_name"] = (
+        names_df["area_name"]
+        .str.replace("county", "")
+        .str.replace("parish", "")
+        .str.strip()
+    )
+    names_df["place_name"] = names_df["place_name"].str.replace("_", " ")
+
+    name_map = names_df.loc[names_df.place_name != names_df.area_name].set_index(
+        "county_id_fips"
+    )
+    name_map = name_map.rename(
+        columns={"place_name": "place_name_old", "area_name": "place_name"}
+    )
+    logger.info(
+        f"Replacing {len(name_map.dropna(how='any'))} county names with data from the latest Census."
+    )
+    logger.info(
+        f"The following place names are getting replaced:\n{name_map.dropna(how='any')}"
+    )
+    combined_df = combined_df.set_index("county_id_fips")
```

Comment:
Does `set_index` handle null values gracefully?

Reply:
In theory `combined_df` still has the lakes in it, which have null `county_id_fips`, yeah?

But I was indeed asking about whether pandas would explode, so it sounds like it's fine!